components/root/root.jsx

```diff
@@ -219,20 +219,16 @@ export default class Root extends React.Component {
         }
     }
 
-    redirectIfNecessary = (props) => {
-        if (props.location.pathname === '/') {
-            if (this.props.noAccounts) {
-                this.props.history.push('/signup_user_complete');
-            } else if (props.showTermsOfService) {
-                this.props.history.push('/terms_of_service');
+    componentDidUpdate(prevProps) {
+        if (this.props.location.pathname === '/') {
+            if (prevProps.noAccounts) {
```

Comment:
The previous behaviour here was actually incorrect. Can you actually change this to the following?
```suggestion
            if (this.props.noAccounts) {
```